lib/models/device.coffee

```diff
@@ -1917,6 +1917,170 @@ getDeviceModel = (deps, opts) ->
+	# @name isTrackingApplicationRelease
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @fulfil {Boolean} - is tracking the current application release
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.isTrackingApplicationRelease('7cf02a6).then(function(isEnabled) {
+	# 	console.log(isEnabled);
+	# });
+	#
+	# @example
+	# resin.models.device.isTrackingApplicationRelease('7cf02a6, function(error, isEnabled) {
+	# 	console.log(isEnabled);
+	# });
+	###
+	exports.isTrackingApplicationRelease = (uuidOrId, callback) ->
+		exports.get(uuidOrId, $select: 'should_be_running__release')
+		.then ({ should_be_running__release }) ->
+			return not should_be_running__release
+		.asCallback(callback)
+
+	###*
+	# @summary Get the hash of the currently tracked release for a specific device
+	# @name getTrackedReleaseHash
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @fulfil {String} - The release hash of the currently tracked release
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.getTrackedReleaseHash('7cf02a6).then(function(release) {
+	# 	console.log(release);
+	# });
+	#
+	# @example
+	# resin.models.device.getTrackedReleaseHash('7cf02a6, function(release) {
+	# 	console.log(release);
+	# });
+	###
+	exports.getTrackedReleaseHash = (uuidOrId, callback) ->
+		exports.get(uuidOrId,
+			$select: 'id'
+			$expand:
+				should_be_running__release:
+					$select: 'commit'
+				belongs_to__application:
+					$select: 'commit'
+		)
+		.then ({ should_be_running__release, belongs_to__application }) ->
+			if not isEmpty(should_be_running__release)
+				return should_be_running__release[0].commit
+			belongs_to__application[0].commit
+		.asCallback(callback)
+
+	###*
+	# @summary Set a specific device to run a particular release
+	# @name setToRelease
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @description Configures the device to run a particular release
+	# and not get updated when the current application release changes.
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @param {String} fullReleaseHash - the hash of a successful release (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.setToRelease('7cf02a6, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.device.setToRelease(123, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.device.setToRelease('7cf02a6, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847', function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.setToRelease = (uuidOrId, fullReleaseHashOrId, callback) ->
+		Promise.try ->
+			if isId(uuidOrId) and isId(fullReleaseHashOrId)
```

Comment:
This will be the branch that the UI will be using for the group action, so that it will need to do as few requests as possible.